### ISSUE TITLE
Rule Request: `explicit_default_argument`

### ISSUE THREAD

**Issue by KacperCzapp** [realm/SwiftLint#5416]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

I would like to propose a new rule `explicit_default_argument` which would trigger a violation if an argument passed to a function or initializer matches the default value provided by the function (or init).

Given the following code:
```swift
struct Foo {
    let bar: Bool
    let baz: String

    init(bar: Boo, baz: String = "") { ... }
    func doSomething(with arg: Bool = true) { ... }
}
```
When calling:
```swift
let foo = Foo(bar: false, baz: "")
foo.doSomething(with: true)
```
Would trigger violations in both lines as they are called with the same values as their defaults. Calling `Foo(bar: false, baz: "baz")` would not trigger the rule. 

This allows for cleaner, more concise code on the call site, especially if there are many default parameters.

> Should the rule be configurable, if so what parameters should be configurable?

If any, I suggest adding two params, for excluding initializers and excluding functions, both set to `false` by default.

> Should the rule be opt-in or enabled by default? Why?

My suggestion is to set the rule as opt-in. It's a style enhancement, not a functional one.

**Comment by SimplyDanny** [comment]
This requires to implement an Analyzer rule, because the rule needs to access the method/initializer declaration from the call-side to figure out the default parameter values. This is impossible on a syntax-level only.
